Doc/reference/expressions.rst

```diff
@@ -83,17 +83,78 @@ exception.
    pair: name; mangling
    pair: private; names
 
-**Private name mangling:** When an identifier that textually occurs in a class
-definition begins with two or more underscore characters and does not end in two
-or more underscores, it is considered a :dfn:`private name` of that class.
-Private names are transformed to a longer form before code is generated for
-them.  The transformation inserts the class name, with leading underscores
-removed and a single underscore inserted, in front of the name.  For example,
-the identifier ``__spam`` occurring in a class named ``Ham`` will be transformed
-to ``_Ham__spam``.  This transformation is independent of the syntactical
-context in which the identifier is used.  If the transformed name is extremely
-long (longer than 255 characters), implementation defined truncation may happen.
-If the class name consists only of underscores, no transformation is done.
+.. rubric:: Private name mangling
+
+When an identifier that textually occurs in a class definition begins with two
+or more underscore characters and does not end in two or more underscores, it
+is considered a :dfn:`private name` of that class.
+
+More precisely, private names are transformed to a longer form before code is
+generated for them.  If the transformed name is longer than 255 characters,
+implementation-defined truncation may happen.
+
+The transformation is independent of the syntactical context in which the
+identifier is used and the transformation rule is defined as follows:
+
+- If the class name consists only of underscores, the transformation is the
+  identity, e.g., the identifier ``__spam`` occurring in a class named ``_``
+  or ``__`` is left as is.
+
+- Otherwise, the transformation inserts the class name, with leading
+  underscores removed and a single underscore inserted, in front of
+  the identifier, e.g., the identifier ``__spam`` occurring in a class
+  named ``Foo``, ``_Foo`` or ``__Foo`` is transformed to ``_Foo__spam``.
+
+  For identifiers declared using :keyword:`import` statements, this rule is
+  slightly different. Indeed, importing a module with a private name directly
+  in a class body raises a :exc:`ModuleNotFoundError` (unless the class name
+  only consists of underscores), as illustrated by the following example:
+
+  .. code-block:: python
+
+     class Foo:
+         import __spam  # raises ModuleNotFoundError at runtime
```

Comment:
Also note that this works if there's a module named `_Foo__spam`.

```suggestion
         import __spam  # tries to import _Foo__spam
```

Reply:
> Also note that this works if there's a module named _Foo__spam.

I feel that I need a separate section for importing private names, so that I can also put the file structure of the example.